src/libslic3r/Fill/FillTpmsFK.cpp

```diff
@@ -358,96 +121,48 @@ void FillTpmsFK::_fill_surface_single(const FillParams&              params,
+    // Density (field period) adjusted to have a good %of weight.
     const float vari_T = 4.18f * spacing * params.multiline / density_factor;
 
-    BoundingBox bb      = expolygon.contour.bounding_box();
-    auto        cenpos  = unscale(bb.center());
-    auto        boxsize = unscale(bb.size());
-    float       xlen    = boxsize.x();
-    float       ylen    = boxsize.y();
-
-    const float delta = 0.4f; // mesh step (adjust for quality/performance)
-    float myperiod = 2 * PI / vari_T;
-    float c_z      = myperiod * this->z; // z height
-
-    // scalar field Fischer-Koch
-    auto scalar_field = [&](float x, float y) -> float {
-        const float a_x       = myperiod * x;
-        const float b_y       = myperiod * y;
-
-        // Fischer - Koch S equation:
-        // cos(2x)sin(y)cos(z) + cos(2y)sin(z)cos(x) + cos(2z)sin(x)cos(y) = 0
-        return cosf(2 * a_x) * sinf(b_y) * cosf(c_z)
-             + cosf(2 * b_y) * sinf(c_z) * cosf(a_x) 
-             + cosf(2 * c_z) * sinf(a_x) * cosf(b_y);
-    };
-
-    // Mesh generation
-    std::vector<std::vector<MarchingSquares::Point>> posxy;
-    int                                              i = 0, j = 0;
-    for (float y = -(ylen) / 2.0f - 0.5f; y < (ylen) / 2.0f + 0.5f; y = y + delta, i++) {
-        j = 0;
-        std::vector<MarchingSquares::Point> colposxy;
-        for (float x = -(xlen) / 2.0f - 0.5f; x < (xlen) / 2.0f + 0.5f; x = x + delta, j++) {
-            MarchingSquares::Point pt;
-            pt.x = cenpos.x() + x;
-            pt.y = cenpos.y() + y;
-            colposxy.push_back(pt);
-        }
-        posxy.push_back(colposxy);
-    }
-
-    std::vector<std::vector<double>> data(posxy.size(), std::vector<double>(posxy[0].size()));
-
-    int width      = posxy[0].size();
-    int height     = posxy.size();
-    int total_size = (height) * (width);
-    tbb::parallel_for(tbb::blocked_range<size_t>(0, total_size),
-                      [&width, &scalar_field, &data, &posxy](const tbb::blocked_range<size_t>& range) {
-                          for (size_t k = range.begin(); k < range.end(); ++k) {
-                              int i      = k / (width);
-                              int j      = k % (width);
-                              data[i][j] = scalar_field(posxy[i][j].x, posxy[i][j].y);
-                          }
-                      });
-
-
-    Polylines polylines;
-    const double contour_value = 0; // offset from theoretical surface
-    MarchingSquares::drawContour(contour_value, width , height , data, posxy, polylines, params);
+    BoundingBox bbox = expolygon.contour.bounding_box();
+    // Slightly enlarge the bounding box to avoid artifacts at the edges.
+    bbox.offset(scale_(1.));
+    marchsq::ScalarField sf = marchsq::ScalarField(bbox, this->z, vari_T);
+    // Get simplified lines using coarse tolerance of 0.1mm (this is infill).
+    Polylines polylines = marchsq::get_polylines(sf, scale_(0.1));
```

Comment:
I tried several values and I think this would be a good balance, with quality similar to current gyroid infill.
```suggestion
    Polylines polylines = marchsq::get_polylines(sf, scale_(0.04));
```

Reply:
Interestingly, you have settled on the same value as `SPARSE_INFILL_RESOLUTION` defined in `libslic3r.h` which I've since discovered. This is probably the correct value to use for a bunch of reasons, so I'll change it to use that.

The other thing I was looking at is the number of lines of gcode generated, and the simplification has a big impact on that because reduces the number of line-segments. I did err a bit more on the side of agressively simplifying things for that reason, but I do think using the defined resolution constant is the right thing to do.